pyro/ops/stats.py

```diff
@@ -1,18 +1,21 @@
 from __future__ import absolute_import, division, print_function
 
+import numbers
+
 import torch
 
 
 def _compute_chain_variance_stats(input):
     # compute within-chain variance and variance estimator
     # input has shape N x C x sample_shape
     N = input.size(0)
-    chain_mean = input.mean(dim=0)
-    var_between = chain_mean.var(dim=0)
-
     chain_var = input.var(dim=0)
     var_within = chain_var.mean(dim=0)
-    var_estimator = (N - 1) / N * var_within + var_between
+    var_estimator = (N - 1) / N * var_within
+    if input.size(1) > 1:
+        chain_mean = input.mean(dim=0)
+        var_between = chain_mean.var(dim=0)
+        var_estimator = var_estimator + var_between
```

Comment:
Different from arviz, in Stan, they allow computing n_eff with [num_chains = 1](https://github.com/stan-dev/stan/blob/develop/src/stan/analyze/mcmc/compute_effective_sample_size.hpp#L49).